components/fxa-client/android/src/main/java/mozilla/appservices/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,153 @@
+     * @return [AccessTokenInfo] that stores the token, along with its scope, key and
+     *                           expiration timestamp (in seconds) since epoch when complete
+     */
+    abstract suspend fun getAccessToken(singleScope: String): AccessTokenInfo?
+
+    /**
+     * Fetches the profile object for the current client either from the existing cached state
+     * or from the server (requires the client to have access to the profile scope).
+     *
+     * @param ignoreCache Fetch the profile information directly from the server
+     * @return Profile (optional, if successfully retrieved) representing the user's basic profile info
+     */
+    abstract suspend fun getProfile(ignoreCache: Boolean = false): Profile?
+
+    /**
+     * Fetches the token server endpoint, for authentication using the SAML bearer flow.
+     *
+     * @return Token server endpoint URL string, `null` if it couldn't be obtained.
+     */
+    abstract suspend fun getTokenServerEndpointURL(): String?
+
+    /**
+     * Get the pairing URL to navigate to on the Authority side (typically a computer).
+     *
+     * @return The URL to show the pairing user
+     */
+    abstract fun getPairingAuthorityURL(): String
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID, if available. `null` otherwise.
+     */
+    abstract fun getCurrentDeviceId(): String?
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token, if available. `null` otherwise.
+     */
+    abstract fun getSessionToken(): String?
+
+    /**
+     * Indicates if account needs to be re-authenticated via [beginAuthentication].
+     * Most common reason for an account to need re-authentication is a password change.
+     *
+     * TODO this may return a false-positive, if we're currently going through a recovery flow.
+     * Prefer to be notified of auth problems via [AccountObserver], which is reliable.
+     *
+     * @return A boolean flag indicating if account needs to be re-authenticated.
+     */
+    abstract fun accountNeedsReauth(): Boolean
+
+    /**
+     * This method should be called when a request made with an OAuth token failed with an
+     * authentication error. It will re-build cached state and perform a connectivity check.
+     *
+     * In time, fxalib will grow a similar method, at which point we'll just relay to it.
+     * See https://github.com/mozilla/application-services/issues/1263
+     *
+     * @param singleScope An oauth scope for which to check authorization state.
+     * @return An optional [Boolean] flag indicating if we're connected, or need to go through
+     * re-authentication. A null result means we were not able to determine state at this time.
+     */
+    abstract suspend fun checkAuthorizationStatus(singleScope: String): Boolean?
```

Comment:
It looks like we should drop the param from this function - a-c only ever passes the profile scope and the equiv function in fxa-client takes no param.

Also wondering if we need the bool return here? The implication is that if this found an auth problem, `accountsNeedsReauth()` would start returning `true` (or in my suggestion above, `getAccountState()` would start returning `AuthenticationProblem`) - which seems a bit clearer and easier to document (ie, after this function you should check the account state via the usual means" seems mildly better?)

Reply:
This method feels weird to me, can we kill it?  The docs say "This method should be called when a request made with an OAuth token failed with an authentication error.".  Why should this be the responsibility of the consumer?  It seems better for the component to make this check and emit a `AccountStateChanged` event if it notices there's an issue.

Reply:
It makes sense to me that this becomes internal.

Reply:
+1 to killing the method and having it internally delt with.

In the future, I can see us re-adding it if more non-firefox scopes get introduced to our component - for example, if Fenix wants to check if it already has Monitor scoped access token, but it doesn't, it wouldn't be an error state. Fenix would just need to make sure the user approves the new scope or force-issue a new access token using the session token.

But in the meantime, it doesn't add much value being an exposed API to Fenix or iOS